nucypher/config/node.py

```diff
@@ -1,106 +1,100 @@
 import contextlib
 import os
 from abc import abstractmethod
-from os.path import abspath, dirname
+from tempfile import TemporaryDirectory
 
-from appdirs import AppDirs
 from itertools import islice
 
-import nucypher
-
-APP_DIRS = AppDirs("nucypher", "NuCypher")
-DEFAULT_CONFIG_ROOT = APP_DIRS.user_data_dir
-
-BASE_DIR = abspath(dirname(dirname(nucypher.__file__)))
-PROJECT_ROOT = abspath(dirname(nucypher.__file__))
+from nucypher.config.constants import DEFAULT_CONFIG_ROOT, DEFAULT_CONFIG_FILE_LOCATION, TEMPLATE_CONFIG_FILE_LOCATION
 
 
 class NodeConfiguration:
 
-    TEMPLATE_INI_FILEPATH = os.path.join(BASE_DIR, 'cli', '.nucypher.ini')
-    DEFAULT_INI_FILEPATH = os.path.join(DEFAULT_CONFIG_ROOT, 'nucypher.ini')
-
-    DEFAULT_KEYRING_ROOT = os.path.join(DEFAULT_CONFIG_ROOT, 'keyring')
-
-    DEFAULT_KNOWN_NODE_DIR = os.path.join(DEFAULT_CONFIG_ROOT, 'known_nodes')
-    DEFAULT_KNOWN_CERTIFICATES_DIR = os.path.join(DEFAULT_KNOWN_NODE_DIR, 'certificates')
-    DEFAULT_KNOWN_METADATA_DIR = os.path.join(DEFAULT_KNOWN_NODE_DIR, 'metadata')
-
     DEFAULT_OPERATING_MODE = 'federated'
 
     class NucypherConfigurationError(RuntimeError):
         pass
 
     def __init__(self,
                  config_root: str = DEFAULT_CONFIG_ROOT,
-                 ini_filepath: str = DEFAULT_INI_FILEPATH,
-                 keyring_dir: str = DEFAULT_KEYRING_ROOT,
-                 known_certificates_dir: str = DEFAULT_KNOWN_CERTIFICATES_DIR,
-                 known_metedata_dir: str = DEFAULT_KNOWN_METADATA_DIR,
-                 known_node_dir: str = DEFAULT_KNOWN_NODE_DIR,
+                 config_file_location: str = DEFAULT_CONFIG_FILE_LOCATION,
                  operating_mode: str = DEFAULT_OPERATING_MODE,
+                 temp: bool = True
                  ) -> None:
 
         #
         # Common
         #
-        self.config_root = config_root
-        self.ini_filepath = ini_filepath
-        self.keyring_dir = keyring_dir
-        self.known_node_dir = known_node_dir
-        self.known_certificates_dir = known_certificates_dir
-        self.known_metadata_dir = known_metedata_dir
+
+        if temp is True:
+            self.temp = True
+            self.temp_dir = TemporaryDirectory()
+            self.config_root = self.temp_dir.name
+        else:
+            self.config_root = config_root
+
+        self.operating_mode = 'federated'
```

Comment:
You probably mean `self.operating_mode = operating_mode`?

Reply:
Yes - Fixed upstream.